pkgs/data/misc/media-types/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib, stdenv, fetchFromGitLab }:
+
+stdenv.mkDerivation rec {
+  pname = "media-types";
+  version = "8.0.0";
+
+  src = fetchFromGitLab {
+    domain = "salsa.debian.org";
+    owner = "debian";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-dl6XKxOF9YT9n/iSvZj1juTqSMaZuOF5i7Eb/VZZDFY";
+  };
+
+  installPhase = ''
+    install -D -m0444 mime.types $out/etc/mime.types
+    install -D -m0444 debian/bug-presubj $out/share/bug/media-types/presubj
+    install -D -m0444 debian/changelog $out/share/doc/media-types/changelog
+    install -D -m0444 debian/copyright $out/share/doc/media-types/copyright
+
+    # Add missing mime-types
+    sed -i $out/etc/mime.types \
+      -e '/^application\/x-wais-source\t\t\tsrc$/aapplication\/x-web-app-manifest+json\t\twebapp' \
+      -e '/^text\/vnd.net2phone.commcenter.command\t\tccc$/atext\/vnd.rim.location.xloc\t\t\txloc' \
+      -e 's/^text\/xml$/text\/xml\t\t\t\t\txsd rng/'
+
+    # Generate nginx mime-types
+    mkdir -p  $out/etc/nginx
+    echo "types {" > $out/etc/nginx/nginx-mime.types
+    cat $out/etc/mime.types | sed -e '/^#.*/d' -e '/^$/d' -e 's/.*/\ \ &;/' -ne '/[^[:space:]][[:space:]]\{1,\}[^[:space:]]/p' >> $out/etc/nginx/nginx-mime.types
+    echo "}" >> $out/etc/nginx/nginx-mime.types
+
+    # Remove duplicate mime-types extensions
+    sed -i $out/etc/nginx/nginx-mime.types \
+      -e 's/application\/x-maker\t\t\t\tfrm maker frame fm fb book fbdoc/application\/x-maker\t\t\t\tmaker frame fb book fbdoc/g' \
```

Comment:
We should generalize this to be update proof.

Reply:
In what way? I made it so that there were no false positives.

Reply:
@SuperSandro2000 allow this variant?:
```
    sed -i $out/etc/nginx/nginx-mime.types \
      -e 's/application\/x-maker\t.*/application\/x-maker\t\t\t\tmaker frame fb book fbdoc;/g' \
      -e 's/application\/x-qgis\t.*/application\/x-qgis\t\t\t\tqgs;/g' \
      -e 's/application\/x-scilab\t.*/application\/x-scilab\t\t\t\tsci;/g' \
      -e 's/audio\/AMR\t.*/audio\/AMR\t\t\t\t\tamr;/g' \
      -e 's/audio\/AMR-WB\t.*/audio\/AMR-WB\t\t\t\t\tawb;/g' \
      -e 's/audio\/EVRC-QCP\t.*/audio\/EVRC-QCP\t\t\t\tqcp;/g' \
      -e '/chemical\/x-chemdraw\t.*/d' \
      -e '/chemical\/x-cif\t.*/d' \
      -e '/chemical\/x-cml\t.*/d' \
      -e 's/chemical\/x-mdl-sdfile\t.*/chemical\/x-mdl-sdfile\t\t\tsd;/g' \
      -e 's/chemical\/x-mopac-input\t.*/chemical\/x-mopac-input\t\t\tmop mopcrt zmt;/g' \
      -e 's/chemical\/x-ncbi-asn1-binary\t.*/chemical\/x-ncbi-asn1-binary\t\t\tval;/g' \
      -e '/chemical\/x-ncbi-asn1-spec\t.*/d' \
      -e '/chemical\/x-pdb\t.*/d' \
      -e 's/image\/vnd.globalgraphics.pgb\t.*/image\/vnd.globalgraphics.pgb\t\t\tpgb;/g' \
      -e '/image\/x-corelphotopaint\t.*/d' \
      -e '/image\/x-jg\t.*/d' \
      -e '/audio\/x-gsm\t.*/d' \
      -e '/text\/x-csh\t.*/d' \
      -e '/text\/x-sh\t.*/d' \
      -e 's/text\/x-tcl\t.*/text\/x-tcl\t\t\t\t\ttk;/g'

```

Reply:
I don't have an variant on how to automate checking. I manually checked nginx configuration and removed duplicates.

Reply:
@SuperSandro2000 replaced to patch file.